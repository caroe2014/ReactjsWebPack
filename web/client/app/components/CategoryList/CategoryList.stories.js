// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import CategoryList from '.';

storiesOf('CategoryList', module)
  .add('With Images', () => {
    const categoryList = [{
      image: 'https://thumbs.dreamstime.com/b/sandwich-plate-28397223.jpg',
      id: '101',
      name: 'Build Your Own Sandwich'
    },
    {
      image: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSaoN_zozVlVN4k6jJlaQOAcr1UbBFA7IJIvxXHjQ5x7o58H-zW6w', // eslint-disable-line max-len
      id: '102',
      name: 'Specials'
    },
    {
      image: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSE4FJOwC6UooDbfKebGZngjvGCC4YQn3XDDrJF-LAJ2hRbv-8V_A', // eslint-disable-line max-len
      id: '103',
      name: 'Sides'
    }
    ];
    return (
      <CategoryList keyProps={categoryList} />
    );
  })
  .add('Without Images', () => {
    const categoryList = [{
      id: '201',
      name: 'Build Your Own Sandwich'
    },
    {
      id: '202',
      name: 'Specials'
    },
    {
      id: '203',
      name: 'Sides'
    }];
    return (
      <CategoryList keyProps={categoryList} />
    );
  });
