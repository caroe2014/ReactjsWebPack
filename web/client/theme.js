// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
