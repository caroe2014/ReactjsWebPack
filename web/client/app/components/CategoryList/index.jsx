// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import PropTypes from 'prop-types';
import { Flex, Heading } from 'rebass';
import Tile from 'web/client/app/components/Tile';
import { Trans } from 'react-i18next';
import i18n from 'web/client/i18n';

const Container = styled(Flex)`
  flex-direction: column;
  margin: 120px auto 0px auto;
  height: fit-content;
  max-width: 75em;
  width: 100%;
  padding: 0px;
  &:after{
    content: '';
    margin: auto;
    width: 100%;
    max-width: 570px;
    }
  ${props => props.theme.mediaBreakpoints.mobile`
    justify-content: center;
    padding-left: 10px;
    padding-right: 10px;
  `}
  ${props => props.theme.mediaBreakpoints.tablet`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.desktop`
    justify-content: space-between;
    padding-left: 20px;
    padding-right: 20px;
  `}
`;

const ListContainer = styled(Flex)`
  flex-direction: column;
  flex-flow: row wrap;
  height: fit-content;
  max-width: 75em;
  width: 100%;
  padding: 0px;
  &:after{
    content: '';
    margin: auto;
    width: 100%;
    max-width: 570px;
    }
  ${props => props.theme.mediaBreakpoints.mobile`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.tablet`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.desktop`
    justify-content: space-between;
  `}
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const Toptext = styled(Heading)`
  width: 100%;
  color: ${props => props.theme.colors.primaryTextColor};
  font-size: 18px;
  margin-top: 20px;
  margin-bottom: 20px;
  font-weight: 700;
  ${props => props.theme.mediaBreakpoints.tablet`margin: 20px auto; max-width: 570px;`}
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

class CategoryList extends Component {

  render () {
    const { keyProps, selectCategory, digitalMenuId } = this.props;
    return (
      <ThemeProvider theme={theme}>
        <Container className='category-list' p={[1, 2, 3, 4]}>
          <Toptext className='select-category-header' role='header' tabIndex={0}
            aria-label={i18n.t('CATEGORY_LIST_PAGE_LABEL')}>
            {!digitalMenuId && <Trans i18nKey='CATEGORY_LIST_PAGE_LABEL' />}</Toptext>
          <ListContainer className='category-list-container'
            tabIndex={keyProps.length > 1 ? 0 : -1}
            aria-label={`${(keyProps.length > 1)
              ? `${i18n.t('CATEGORY_SELECT')} ${keyProps.length} ${i18n.t('CATEGORIES')}` : ``}`}>
            {keyProps && keyProps.map((item, index) => (
              <React.Fragment key={item.id}>
                <Tile
                  classContext={`tile category-${index + 1} category_${item.id}`}
                  imageresponsive='true'
                  className='tile'
                  title={item.name}
                  image={item.image}
                  id={item.id}
                  handleClick={selectCategory}
                  ariaLabel={`category${index + 1} ${item.name}`}
                  tabIndex={0}
                />
              </React.Fragment>
            ))}
          </ListContainer>
        </Container>
      </ThemeProvider>
    );
  }
}

CategoryList.propTypes = {
  selectCategory: PropTypes.func,
  keyProps: PropTypes.arrayOf(
    PropTypes.shape({
      image: PropTypes.string,
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    }).isRequired
  )
};
export default CategoryList;
