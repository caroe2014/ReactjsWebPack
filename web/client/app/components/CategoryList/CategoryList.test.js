// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import CategoryList from '.';
import { t } from 'i18next';

configure({ adapter: new Adapter() });

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

jest.mock('i18next');
t.mockImplementation((key) => {
  return key;
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('CategoryList', () => {

  let wrapper;
  let mockProps = [
    {
      image: 'https://thumbs.dreamstime.com/b/sandwich-plate-28397223.jpg',
      id: '101',
      name: 'Build Your Own Sandwich'
    },
    {
      image: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSaoN_zozVlVN4k6jJlaQOAcr1UbBFA7IJIvxXHjQ5x7o58H-zW6w', // eslint-disable-line max-len
      id: '102',
      name: 'Specials'
    },
    {
      image: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSE4FJOwC6UooDbfKebGZngjvGCC4YQn3XDDrJF-LAJ2hRbv-8V_A', // eslint-disable-line max-len
      id: '103',
      name: 'Sides'
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<CategoryList keyProps={mockProps} />
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render without throwing an error ob mount', () => {
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <CategoryList keyProps={mockProps}/>
      </ThemeProvider>
    );
    expect(mountWrapper.exists()).toBe(true);
  });

  describe('when Item props are received', () => {
    it('should render the same number of tiles', () => {
      expect(wrapper.find('.tile').length).toBe(mockProps.length);
    });
  });

  describe('when Item props are not received', () => {
    beforeEach(() => {
      mockProps = [];
    });
    it('should not render CategoryList', () => {
      const itemWrapper = shallow(<CategoryList keyProps={mockProps} />);
      expect(itemWrapper.find('.tile').length).toBe(0);
    });
  });

});
