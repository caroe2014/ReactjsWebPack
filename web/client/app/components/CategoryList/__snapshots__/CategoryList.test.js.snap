// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList does a snapshot test 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <Styled(Flex)
    className="category-list"
    p={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
  >
    <Styled(Heading)
      aria-label="CATEGORY_LIST_PAGE_LABEL"
      className="select-category-header"
      role="header"
      tabIndex={0}
    >
      <div
        i18nKey="CATEGORY_LIST_PAGE_LABEL"
      />
    </Styled(Heading)>
    <Styled(Flex)
      aria-label="CATEGORY_SELECT 3 CATEGORIES"
      className="category-list-container"
      tabIndex={0}
    >
      <Tile
        ariaLabel="category1 Build Your Own Sandwich"
        classContext="tile category-1 category_101"
        className="tile"
        hasBorder={true}
        id="101"
        image="https://thumbs.dreamstime.com/b/sandwich-plate-28397223.jpg"
        imageresponsive="true"
        isEnabled={true}
        tabIndex={0}
        title="Build Your Own Sandwich"
      />
      <Tile
        ariaLabel="category2 Specials"
        classContext="tile category-2 category_102"
        className="tile"
        hasBorder={true}
        id="102"
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSaoN_zozVlVN4k6jJlaQOAcr1UbBFA7IJIvxXHjQ5x7o58H-zW6w"
        imageresponsive="true"
        isEnabled={true}
        tabIndex={0}
        title="Specials"
      />
      <Tile
        ariaLabel="category3 Sides"
        classContext="tile category-3 category_103"
        className="tile"
        hasBorder={true}
        id="103"
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSE4FJOwC6UooDbfKebGZngjvGCC4YQn3XDDrJF-LAJ2hRbv-8V_A"
        imageresponsive="true"
        isEnabled={true}
        tabIndex={0}
        title="Sides"
      />
    </Styled(Flex)>
  </Styled(Flex)>
</ThemeProvider>
`;
